### PR TITLE
SWITCHYARD-2836 - fixing the other half of this equation (will update

### PR DESCRIPTION
**Pull request by bfitzpat**
the JIRA with details)

### CODE REVIEW

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java [hunk 1]
```diff
@@ -172,6 +172,14 @@ public boolean update(IUpdateContext context) {
                 for (ComponentReference componentReference : component.getReference()) {
                     if (reference.getName().equals(componentReference.getName())) {
                         promotedReferences.add(componentReference);
+
+                        // fix the promote issue if we can here for SWITCHYARD-2836
+                        if (reference.getPromote().isEmpty()) {
+                            if (reference.getInterface().eClass().getName().
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java]: we are sure that reference.getInterface() is not null and componentReference.getInterface() either?

Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java]: In both cases, the Component and Composite references (unless they've been destroyed in a text editor) will have an interface, so yes - with about 90% certainty I'm going to say yes. The tooling won't allow anything else.

Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java]: the tooling yes but user can modify the source code directly, isn't it?

Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java]: not in the SwitchYard editor. they would have to open the config file in the text or XML editor and edit it by hand, which to my mind invalidates any security they would have gotten from editing via the SY editor. The source tab in the SY editor is read only so they are unable to modify it there

Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java]: ok

**Comment by apupier** [comment]
+1

**Comment by bfitzpat** [comment]
Closing this one as we are no longer planning any Mars updates. Can re-open if we need to down the line.

**Pull request closed without merging**